File: apps/admin_mobile/lib/application/use_cases/reservations/transition_reservation_use_case.dart
import '../../../domain/entities/reservation.dart';
import '../../../domain/repositories/reservation_repository.dart';

class TransitionReservationUseCase {
  final ReservationRepository _repo;
  TransitionReservationUseCase(this._repo);

  Future<Reservation> call(int id, ReservationAction action) =>
      _repo.transition(id, action);
}

File: apps/admin-v2/src/application/use-cases/reservations/transition-reservation.use-case.ts
import type { ReservationRepository } from '@/domain/repositories/reservation.repository';
import type { ReservationAction } from '@/domain/entities/reservation';

export class TransitionReservationUseCase {
  constructor(private repo: ReservationRepository) {}

  execute(id: string, action: ReservationAction) {
    return this.repo.transition(id, action);
  }
}
